src/components/floating-emojis/FloatingEmojis.tsx

```diff
@@ -0,0 +1,151 @@
+import React, { useCallback, useEffect, useMemo, useState, ReactNode } from 'react';
+import { Animated, View, ViewProps } from 'react-native';
+import FloatingEmoji from './FloatingEmoji';
+import GravityEmoji from './GravityEmoji';
+import { useTimeout } from '@/hooks';
+import { position } from '@/styles';
+import { DebugLayout } from '@/design-system';
+import { DEVICE_HEIGHT, DEVICE_WIDTH } from '@/utils/deviceUtils';
+import { AbsolutePortal } from '../AbsolutePortal';
+
+interface Emoji {
+  emojiToRender: string;
+  x: number;
+  y: number;
+}
+
+interface FloatingEmojisProps extends Omit<ViewProps, 'children'> {
+  centerVertically?: boolean;
+  children?: ReactNode | ((props: { onNewEmoji: (x?: number, y?: number) => void }) => ReactNode);
+  disableHorizontalMovement?: boolean;
+  disableRainbow?: boolean;
+  disableVerticalMovement?: boolean;
+  distance?: number;
+  duration?: number;
+  emojis: string[];
+  fadeOut?: boolean;
+  gravityEnabled?: boolean;
+  marginTop?: number;
+  opacity?: number | Animated.AnimatedInterpolation<number>;
+  opacityThreshold?: number;
+  range?: [number, number];
+  scaleTo?: number;
+  setOnNewEmoji?: (fn: ((x?: number, y?: number) => void) | undefined) => void;
+  size: number;
+  wiggleFactor?: number;
+}
+
+const EMPTY_ARRAY: Emoji[] = [];
+const getEmoji = (emojis: string[]) => Math.floor(Math.random() * emojis.length);
+const getRandomNumber = (min: number, max: number) => Math.random() * (max - min) + min;
+
+const FloatingEmojis: React.FC<FloatingEmojisProps> = ({
+  centerVertically,
+  children,
+  disableHorizontalMovement,
+  disableRainbow,
+  disableVerticalMovement,
+  distance = 130,
+  duration = 2000,
+  emojis,
+  fadeOut = true,
+  gravityEnabled,
+  marginTop,
+  opacity = 1,
+  opacityThreshold,
+  range: [rangeMin, rangeMax] = [0, 80],
+  scaleTo = 1,
+  setOnNewEmoji,
+  size = 30,
+  wiggleFactor = 0.5,
+  style,
+  ...props
+}) => {
+  const emojisArray = useMemo(() => (Array.isArray(emojis) ? emojis : [emojis]), [emojis]);
+  const [floatingEmojis, setEmojis] = useState<Emoji[]>(EMPTY_ARRAY);
+  const [startTimeout, stopTimeout] = useTimeout();
+  const clearEmojis = useCallback(() => setEmojis(EMPTY_ARRAY), []);
+
+  // 🚧️ TODO: 🚧️
+  // Clear emojis if page navigatorPosition falls below 0.93 (which we should call like `pageTransitionThreshold` or something)
+  // otherwise, the FloatingEmojis look weird during stack transitions
```

Comment:
is this worth a discussion ticket?

Reply:
this was here previous to this PR, just has my name on it since .js -> .tsx. Fine to open a ticket about it if needed

Reply:
Oops, didn't mean to make you defend your pasta's honor. Shoulda caught it.